Sync docs with website repo (#1567)
grafana/metrictank: docs/cloud/http-api.md

---
title: HTTP API
weight: 2
---

# HTTP API

The HTTP API is the same as that of Graphite, with the addition of ingestion, authentication and meta tags.

First of all, there are two endpoints you will be talking to. They are provided on your grafana.com Hosted Metrics instance details page.

They will look something like:

* `<base_in>` : `https://something.grafana.net/metrics`
* `<base_out>` : `https://something.grafana.net/graphite`

Furthermore, you will need to provision API keys to talk to the API. Each key will be of one of these types:

* Viewer
* MetricsPublisher
* Editor
* Admin

> API keys can be provisioned on your Grafana.com organisation page under "security > API Keys".

Reviewer: ```suggestion
A "Viewer" key can only be used to execute queries, while a "MetricsPublisher" key can only be used to push metrics into the platform.  "Editor" and "Admin" keys can be used for both pushing metrics and performing queries.

API keys can be provisioned on your Grafana.com organization page under "Security > API Keys".
```

## Authentication

Authentication differs depending on whether your instance is provisioned on a dedicated or a shared cluster.
How to tell the difference?

> On your Grafana.com instance details page, if your query and metrics endpoint look generic like this:

Reviewer: ```suggestion
On your Grafana.com instance details page, if your query and metrics endpoint look like this:
```

```
https://graphite-us-central1.grafana.net/graphite
https://graphite-us-central1.grafana.net/metrics
```

Then you are on a shared cluster. However if your URL's look more like this:

```
https://tsdb-123-your-company-name.hosted-metrics.grafana.net/graphite
https://tsdb-123-your-company-name.hosted-metrics.grafana.net/metrics
```

Then you are on a dedicated cluster.

You have several ways to authenticate:

* For dedicated clusters, any of these HTTP headers will work:

```
Authorization: Basic base64(api_key:<api key>)
Authorization: Bearer <api key>
Authorization: Bearer api_key:<api key>
```

* For shared clusters, use any of these HTTP headers:

```
Authorization: Basic base64(<instance id>:<api key>)
Authorization: Bearer <instance id>:<api key>
```

Note that you can find the instance ID as the username in the "Using Grafana with Hosted Metrics" section of your Grafana.com instance details page

So essentially you can use basic auth with username "api_key" (for dedicated clusters) or your instance ID (for shared clusters) and password the api key that you provisoned

> And you can also use a bearer token in "username:password" format (if username not specified, "api_key" is assumed)

Reviewer: ```suggestion

You can also use a bearer token in "username:password" format (if username not specified, "api_key" is assumed)
```
